src/content/blog/content-layer-deep-dive.mdx

```diff
@@ -0,0 +1,230 @@
+```
+
+This data is then available in your `.astro` components.
+
+```astro
+---
+// src/pages/countries/[id].astro
+import type { GetStaticPaths } from 'astro';
+import { getCollection } from 'astro:content';
+
+export const getStaticPaths: GetStaticPaths = async () => {
+	const collection = await getCollection('countries');
+	if (!collection) return [];
+	return collection.map((country) => ({
+		params: {
+			id: country.id,
+		},
+		props: {
+			country,
+		},
+	}));
+};
+
+const { country } = Astro.props;
+---
+
+<h1>{craft.data.name}</h1>
+<p>Capital: {craft.data.capital}</p>
+```
+
+## Content layer lifecycle
+
+When `astro build` or `astro dev` is run, the loader for each collection is invoked. These loaders update their own scoped data store, which is preserved between builds.
+
+<BlogContentImage src={astroBuildImage} />
+
+Astro components and pages can then use `getCollection` or `getEntry` to query the data. The data is immutable at that point, so the pages are all querying the same snapshot, compiled at build time. This applies whether the page is being prerendered at build time, or rendered on-demand in SSR. The important thing to note here is that the data store is only updated at build time: a deployed site cannot change the data store. If a data source needs to update a collection it must do this by triggering a new build.
+
+During development the data store can be updated on demand, by using the hotkey `s+enter` when running `astro dev`. Integrations can also update the data during dev. This could be used for things such as registering a development refresh endpoint, or opening a socket to a CMS to listen for updates. This is just available during development though: once deployed the data store is immutable.
+
+## How a loader works
+
+A loader defines how a collection loads its data. It is executed during `astro build` and `astro dev`, or when the content is refreshed by an integration. The example above shows the simplest kind of loader: an async function that returns an array. However for more complex cases an object loader gives more flexibility because it can directly control the data store.
+
+A loader interacts with the content layer via a data store object. This is a key-value store that is scoped to an individual collection. A collection can only access its own entries, but has complete control over these. If a loader knows that its data source hasn't changed then it can skip updating entirely, or it can just update the entries which have changed. There are some tools that are provided to make this easier. First is the metadata store, which can be used to store arbitrary values such as "last modified" times, or sync tokens. Comparing these will allow the loader to do things like make conditional API requests, or use delta sync APIs. This example shows how to do this with an RSS feed loader. It stores the last modified and etag headers in the metadata store, and then uses those to make conditional requests.
```

Comment:
```suggestion
A loader interacts with the content layer via a data store object. This is a key-value store that is scoped to an individual collection. A collection can only access its own entries, but has complete control over these. If a loader knows that its data source hasn't changed then it can skip updating entirely, or it can just update the entries which have changed.

The Content Layer API provides some tools to make this easier. First is the metadata store, which can be used to store arbitrary values such as "last modified" times, or sync tokens. Comparing these will allow the loader to do things like make conditional API requests, or use delta sync APIs.

This example shows how to do this with an RSS feed loader. It stores the last modified and etag headers in the metadata store, and then uses those to make conditional requests:
```